llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -318,6 +305,56 @@ class RegionEndEdges : public ScheduleDAGMutation {
   };
 };
 
+/// This Mutator is responsible for emitting "fixed" SUnits at the top or bottom
+/// of the region. These special SUnits require a specific cycle and cannot be
+/// placed freely by the scheduler.
+///
+/// Here, these special SUnits get created from Region::top_fixed_instrs() or
+/// Region::bot_fixed_instrs(), and dependencies are created between "free" and
+/// "fixed" SUnits.
```

Comment:
Perhaps spend some words on the mechanism. As I understand it, we only add edges to ExitSU with the right latency, and we make sure that  pickOnlyChoice will not return Free nodes before exhausting the Fixed ones.